x-pack/plugins/security_solution/public/management/pages/event_filters/view/hooks.ts

```diff
@@ -42,3 +49,17 @@ export const useEventFiltersNotification = () => {
     toasts.addDanger(getCreationErrorMessage(creationError));
   }
 };
+
+export type NavigationCallback = (
+  ...args: Parameters<Parameters<typeof useCallback>[0]>
+) => Partial<EventFiltersPageLocation>;
+
+export function useEventFiltersNavigateCallback(callback: NavigationCallback) {
```

Comment:
yeah, I would not recoment using this utility hook and just go with normal react `useCallback()`.